project/src/main/java/com/google/sps/servlets/UserProfileEventsServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.EventVolunteering;
+import com.google.sps.utilities.CommonUtils;
+import com.google.sps.utilities.SpannerTasks;
+import java.io.IOException;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user-events")
+public class UserProfileEventsServlet extends HttpServlet {
+  private static final String EVENT_TYPE = "event-type";
+  private static final String VOLUNTEERING = "volunteering";
+
+  /** 
+   * Gets the current user's events corresponding to the event type specified as a parameter.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String eventType = request.getParameter(EVENT_TYPE);
+    User user = UserServiceFactory.getUserService().getCurrentUser();
+
+    if (user == null) {
+      response.sendRedirect("/index.html");
+      return;
+    } else if (eventType == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, String.format("No event type specified."));
+      return;
+    }
+
+    switch (eventType) {
```

Comment:
Is it possible to break this into a helper method? doGet is getting a little long in terms of different things it's doing

Maybe return the correct Json or throw an exception, and act appropriate in this method

Reply:
Done, yes good idea.